diff -crB a/src/CMakeLists.txt b/src/CMakeLists.txt
*** a/src/CMakeLists.txt	2014-02-05 17:24:11.000000000 +0100
--- b/src/CMakeLists.txt	2015-03-14 23:41:30.706755532 +0100
***************
*** 24,30 ****
  add_library(ftdi1 SHARED ${c_sources})
  
  math(EXPR VERSION_FIXUP "${MAJOR_VERSION} + 1")    # Compatiblity with previous releases
! set_target_properties(ftdi1 PROPERTIES VERSION ${VERSION_FIXUP}.${MINOR_VERSION}.0 SOVERSION 2)
  # Prevent clobbering each other during the build
  set_target_properties ( ftdi1 PROPERTIES CLEAN_DIRECT_OUTPUT 1 )
  
--- 24,31 ----
  add_library(ftdi1 SHARED ${c_sources})
  
  math(EXPR VERSION_FIXUP "${MAJOR_VERSION} + 1")    # Compatiblity with previous releases
! # Disable shared library versioning for ANDROID
! # set_target_properties(ftdi1 PROPERTIES VERSION ${VERSION_FIXUP}.${MINOR_VERSION}.0 SOVERSION 2)
  # Prevent clobbering each other during the build
  set_target_properties ( ftdi1 PROPERTIES CLEAN_DIRECT_OUTPUT 1 )
  
